src/providers/push/fcm/fcm.provider.ts

```diff
@@ -0,0 +1,157 @@
+import {inject, Provider} from '@loopback/core';
+import {HttpErrors} from '@loopback/rest';
+import * as admin from 'firebase-admin';
+import {ConditionMessage, TopicMessage} from 'firebase-admin/lib/messaging';
+import {FcmBindings} from './keys';
+import {
+  FcmConfig,
+  FcmMessage,
+  FcmNotification,
+  FcmSubscriberType,
+} from './types';
+
+export class FcmProvider implements Provider<FcmNotification> {
+  constructor(
+    @inject(FcmBindings.Config, {
+      optional: true,
+    })
+    private readonly fcmConfig?: FcmConfig,
+  ) {
+    if (this.fcmConfig) {
+      this.fcmService = admin.initializeApp({
+        credential: admin.credential.cert(this.fcmConfig.serviceAccountPath),
+        databaseURL: this.fcmConfig.dbUrl,
+      });
+    } else {
+      throw new HttpErrors.PreconditionFailed('Firebase Config missing !');
+    }
+  }
+
+  fcmService: admin.app.App;
+
+  value() {
+    return {
+      publish: async (message: FcmMessage) => {
```

Comment:
this is one big function. Can't we move it to smaller functions and call them here?